### PR TITLE
Refactor Storage for ingredient and recipe

### LINKED ISSUE #44: Missing storage for Recipe and Ingredient

### PR DESCRIPTION
**Pull request by Trav1sT**
Closes #44

### CODE REVIEW

File: src/main/java/chopchop/storage/StorageManager.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    @Override
+    public Optional<ReadOnlyRecipeBook> readRecipeBook(Path filePath)
+        throws DataConversionException, IOException {
+        logger.fine("Attempting to read data from file: " + filePath);
+        return recipeBookStorage.readRecipeBook(filePath);
+    }
+
+    @Override
+    public Path getRecipeBookFilePath() {
+        return null;
```
Comment (zhiayang) [inline src/main/java/chopchop/storage/StorageManager.java:79]: not done yet ah

File: src/main/java/chopchop/logic/LogicManager.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
+package chopchop.logic;
```
Comment (zhiayang) [inline src/main/java/chopchop/logic/LogicManager.java:1]: wait a minute, isn't this file a duplicate of the `CommandDispatcher`

File: src/main/java/chopchop/storage/StorageManager.java (hunk 1 above)
Comment (Trav1sT) [reply src/main/java/chopchop/storage/StorageManager.java]: Uwu I just realised this after adding the tests.

File: src/main/java/chopchop/storage/JsonAdaptedRecipe.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import chopchop.model.attributes.Step;
+import chopchop.model.ingredient.IngredientReference;
+import chopchop.model.recipe.Recipe;
+import chopchop.util.Result;
+
+public class JsonAdaptedRecipe {
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Recipe's %s field is missing!";
+
+    private final String name;
+    private final List<String> ingredientRefs;
```
[the remark this answers is not in the document]
Comment (Trav1sT) [reply src/main/java/chopchop/storage/JsonAdaptedRecipe.java:19]: Agreed, working on it!

File: src/main/java/chopchop/model/ingredient/IngredientReference.java [hunk 4]
```diff
@@ -20,6 +21,24 @@ public IngredientReference(String name, Quantity qty) {
...
+    public static Result<IngredientReference> parse(String source) {
+        String[] words = source.split(" \\(|\\)");
+        if (words.length != 2) {
+            return Result.error("Unable to parse string: %s", source);
+        }
+        Result<Quantity> qtyResult = Quantity.parse(words[1]);
+        if (qtyResult.isError()) {
+            return Result.error(qtyResult.getError());
+        }
+        return Result.of(new IngredientReference(words[1], qtyResult.getValue()));
```
[the remark this answers is not in the document]
Comment (Trav1sT) [reply src/main/java/chopchop/model/ingredient/IngredientReference.java:39]: Oh yes should be `words[0]`

File: src/main/java/chopchop/logic/LogicManager.java (hunk 2 above)
Comment (Trav1sT) [reply src/main/java/chopchop/logic/LogicManager.java:1]: Deleting this old and deproved version.

File: src/main/java/chopchop/storage/JsonAdaptedRecipe.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhiayang) [reply src/main/java/chopchop/storage/JsonAdaptedRecipe.java:19]: ups

**Comment by Trav1sT** [comment]
Hmmm, I think it's alright.

**Comment by zhiayang** [comment]
ups

File: src/main/java/chopchop/MainApp.java [hunk 5]
```diff
@@ -56,41 +56,55 @@ public void init() throws Exception {
...
         initLogging(config);
 
         model = initModelManager(storage, userPrefs);
 
-        logic = new LogicManager(model, storage);
+        //logic = new LogicManager(model, storage);
 
-        ui = new UiManager(logic);
-    }
+        //ui = new UiManager(logic);
```
Comment (seowalex) [inline src/main/java/chopchop/MainApp.java:70]: Wouldn't this cause the logic and ui to not work?

**Comment by seowalex** [comment]
Other than that it looks ok, though there seems to be a bunch of repeated code which I suspect can be extracted, but we can leave that to 1.3.

File: src/main/java/chopchop/MainApp.java (hunk 5 above)
Comment (Trav1sT) [reply src/main/java/chopchop/MainApp.java:70]: I commented it out cuz LogicManager and UiManager both ain't in chopchop yet. So the red line kind of annoying.

**Comment by zhiayang** [comment]
Looks good to me :+1:

**Pull request merged**